TESTS/netsocket/README.md

```diff
@@ -1654,6 +1669,311 @@ through a dedicated socket
+
+### ASYNCHRONOUS_DNS_TIMEOUTS
+
+**Description:**
+
+Test DNS timeouts using an external event queue that is modified to timeout the events faster that standard event queue. In this test event queue shall not delay events, instead it handles those immediately. Call `NetworkInterface::gethostbyname_async()` in a row 6 times with a different host names. All or some of the request shall timeout and timeout return value is returned.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Define a thread and an event queue running on it.
+2.  Define a DNS call in callback function that uses the event queue (call_in_callback_cb_t). Callback function shall not delay callbacks; instead it shall handle those immediately.
+3.  Start thread and event queue.
+4.  Set DNS callback function using the `nsapi_dns_call_in_set()` call.
+5.  Call `gethostbyname_async()` in a row 6 times with a different host names. Host names shall not be found from cache.
+6.  Verify that last `gethostbyname_async()` operation is rejected since there is room only for 5 simultaneous operations.
+7.  Verify that callback is called with correct parameters 5 times.
+
+**Expected result:**
+
+Sixth `gethostbyname_async()` is rejected. At least for one operation, callback is called with NSAPI_ERROR_TIMEOUT value.
+
+### ASYNCHRONOUS_DNS_SIMULTANEOUS_REPEAT
+
+**Description:**
+
+Verify that simultaneous asynchronous DNS queries work correctly when repeated in sequence. Call `NetworkInterface::gethostbyname_async()` in a row 5 times with a different host names. Wait for all requests to complete and verify result. Repeat the procedure 100 times.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname_async()` in a row 5 times with a different host names.
+2.  Verify that callback is called with correct parameters 5 times for first operation.
+3.  Repeat 100 times steps 1 to 2.
+
+**Expected result:**
+
+Callback is called with NSAPI_ERROR_OK and IP address 5 times for every repeat.
+
+### SYNCHRONOUS_DNS
+
+**Description:**
+
+Verify basic functionality of synchronous DNS. Call `NetworkInterface::gethostbyname()` with a valid host name and verify result.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname()` with a valid host name.
+2.  Verify that address was resolved and return value was valid.
+
+**Expected result:**
+
+Return value is NSAPI_ERROR_OK and IP address is obtained from the function call.
+
+### SYNCHRONOUS_DNS_MULTIPLE
+
+**Description:**
+
+Verify basic functionality of synchronous DNS. Call `NetworkInterface::gethostbyname()` with a list of 6 host names and verify result.
```

Comment:
"Verify the basic functionality of synchronous DNS ... names, and verify the result."